### PR TITLE
Feat di

### PR DESCRIPTION
**Pull request by Kabak-Siarhei**
Added DI for SharedPreference

### CODE REVIEW

File: lib/home/repository/service_locator.dart [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import 'package:get_it/get_it.dart';
+import 'package:word_to_gif/home/blocs/theme/theme_bloc.dart';
+import 'package:word_to_gif/home/repository/shared_preferences.dart';
+
+final sl = GetIt.I;
```
Comment (shcherbuk96) [inline lib/home/repository/service_locator.dart:5]: Rename to getIt

File: lib/home/blocs/theme/theme_bloc.dart [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import 'package:bloc/bloc.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:word_to_gif/home/repository/service_locator.dart';
```
Comment (shcherbuk96) [inline lib/home/blocs/theme/theme_bloc.dart:3]: You should send to the bloc your requested services

File: lib/home/blocs/theme/theme_bloc.dart [hunk 3]
```diff
@@ -20,12 +21,12 @@ class ThemeBloc extends Bloc<ThemeEvent, ThemeState> {
   }
 
   Stream<ThemeState> _initialEvent(_InitialEvent event) async* {
-    final isDarkTheme = await SharedPrefs().getSharedBool(key: themeKey);
+    final isDarkTheme = await sl.get<SharedPrefs>().getSharedBool(key: themeKey);
```
Comment (shcherbuk96) [inline lib/home/blocs/theme/theme_bloc.dart:24]: You should send to the bloc your requested services
To constructor

File: lib/home/blocs/theme/theme_bloc.dart [hunk 4]
```diff
@@ -20,12 +21,12 @@ class ThemeBloc extends Bloc<ThemeEvent, ThemeState> {
...
     yield ThemeState.changeState(isDarkTheme: isDarkTheme);
   }
 
   Stream<ThemeState> _changeEvent(_ChangeEvent event) async* {
-    SharedPrefs().setSharedBool(key: themeKey, val: event.isDarkTheme);
+    sl.get<SharedPrefs>().setSharedBool(key: themeKey, val: event.isDarkTheme);
```
Comment (shcherbuk96) [inline lib/home/blocs/theme/theme_bloc.dart:29]: You should send to the bloc your requested services
To constructor

File: lib/home/repository/service_locator.dart (hunk 1 above)
Comment (Kabak-Siarhei) [reply lib/home/repository/service_locator.dart:5]: done

File: lib/home/blocs/theme/theme_bloc.dart (hunk 3 above)
Comment (Kabak-Siarhei) [reply lib/home/blocs/theme/theme_bloc.dart]: Fixed

File: lib/home/blocs/theme/theme_bloc.dart (hunk 4 above)
Comment (Kabak-Siarhei) [reply lib/home/blocs/theme/theme_bloc.dart]: Fixed

File: lib/home/blocs/theme/theme_bloc.dart [hunk 5]
```diff
@@ -20,12 +21,12 @@ class ThemeBloc extends Bloc<ThemeEvent, ThemeState> {
...
 
   Stream<ThemeState> _initialEvent(_InitialEvent event) async* {
-    final isDarkTheme = await SharedPrefs().getSharedBool(key: themeKey);
+    final isDarkTheme = await getIt.get<SharedPrefs>().getSharedBool(key: themeKey);
     yield ThemeState.changeState(isDarkTheme: isDarkTheme);
   }
 
   Stream<ThemeState> _changeEvent(_ChangeEvent event) async* {
-    SharedPrefs().setSharedBool(key: themeKey, val: event.isDarkTheme);
+    getIt.get<SharedPrefs>().setSharedBool(key: themeKey, val: event.isDarkTheme);
```
Comment (shcherbuk96) [inline lib/home/blocs/theme/theme_bloc.dart:29]: Nope. You should send sharedprefs to bloc constructor

File: lib/home/repository/service_locator.dart (hunk 1 above)
Comment (shcherbuk96) [reply lib/home/repository/service_locator.dart]: > 

Where?

File: lib/home/blocs/theme/theme_bloc.dart (hunk 3 above)
Comment (shcherbuk96) [reply lib/home/blocs/theme/theme_bloc.dart]: Where?

File: lib/home/blocs/theme/theme_bloc.dart (hunk 4 above)
Comment (shcherbuk96) [reply lib/home/blocs/theme/theme_bloc.dart]: Where?

Comment (Kabak-Siarhei) [reply lib/home/blocs/theme/theme_bloc.dart]: I've fixed it in a another tree.

File: lib/home/blocs/theme/theme_bloc.dart (hunk 3 above)
Comment (Kabak-Siarhei) [reply lib/home/blocs/theme/theme_bloc.dart]: I've fixed it in a another tree.

File: lib/home/repository/service_locator.dart (hunk 1 above)
Comment (Kabak-Siarhei) [reply lib/home/repository/service_locator.dart]: I've fixed it in a another tree.

**Pull request merged**